py_gtfs_rt_ingestion/lib/s3_utils.py

```diff
@@ -132,47 +135,74 @@ def _init_process_session() -> None:
     ].resource("s3")
 
 
-def move_s3_objects(files: List[str], to_bucket: str) -> None:
+def move_s3_objects(files: List[str], to_bucket: str) -> List[str]:
     """
     Move list of S3 objects to to_bucket bucket, retaining the object path.
 
     :param files: list of s3 filepath uris
     :param destination: directory or S3 bucket to move to formatted without
         leading 's3://'
 
-    No return value.
+    :reutrn - list of 3s objects that failed to move
     """
     to_bucket = to_bucket.split("/")[0]
 
+    files_to_move = set(files)
+    found_exception = Exception("No exception reported from s3 move pool.")
+
     # this is the default pool size for a ThreadPoolExecutor as of py3.8
     cpu_count = cast(int, os.cpu_count() if os.cpu_count() is not None else 1)
     # make sure each pool will have at least 50 files to move
     files_per_pool = 50
-    max_pool_size = max(1, int(len(files) / files_per_pool))
-    pool_size = min(32, cpu_count + 4, max_pool_size)
+    # retry attempts on failed file moves
+    retry_count = 3
 
     process_logger = ProcessLogger(
         "move_s3_objects",
-        pool_size=pool_size,
         to_bucket=to_bucket,
         file_count=len(files),
     )
     process_logger.log_start()
 
-    error_count = 0
-    with Pool(pool_size, initializer=_init_process_session) as pool:
-        for pool_error_count in pool.starmap(
-            _move_s3_object, [(filename, to_bucket) for filename in files]
-        ):
-            error_count += pool_error_count
+    for retry_attempt in range(retry_count):
+        max_pool_size = max(1, int(len(files_to_move) / files_per_pool))
+        pool_size = min(32, cpu_count + 4, max_pool_size)
+        process_logger.add_metadata(pool_size=pool_size)
+        try:
+            with Pool(pool_size, initializer=_init_process_session) as pool:
+                for file_moved in pool.starmap(
+                    _move_s3_object,
+                    [(filename, to_bucket) for filename in files_to_move],
+                ):
+                    # remove filename from set if moved without exception
+                    if isinstance(file_moved, str):
+                        files_to_move.discard(file_moved)
+
+        except Exception as exception:
+            found_exception = exception
+
+        # all files moved, exit retry loop
+        if len(files_to_move) == 0:
+            break
+
+        # wait for gremlins to disappear
+        time.sleep(15)
```

Comment:
retry logic for s3 move operation. also enclose pool creation in try/catch block, failure could occur during initialization of s3 client